### DISCUSSION TITLE
Volume control for Nvidia Shield

### DISCUSSION THREAD

**Discussion by PRProd** [PRProd/HA-Firemote#490] [Q&A]
Question from @unc2k3 copied from https://github.com/PRProd/HA-Firemote/issues/489#issuecomment-2316243176
<hr>
As a side note, I've also got 2 Nvidia Sheilds. One works flawlessly and the other works with the exception of the volume buttons. Any idea what that might be? I've tried both ADB entities and switched between Remote and TV for the Associated Android TV Remote entity. Only the "Remote" gives me any functionality, but still not volume controls either way.

Here's the YAML:
type: custom:firemote-card
entity: media_player.android_tv_192_168_3_46
device_family: nvidia-shield
device_type: shield-tv-2019
compatibility_mode: default
android_tv_remote_entity: remote.master_tv
visible_name_text: Master TV
defaultRemoteStyle_override: NS2
app_launch_1: tivimate
app_launch_2: youtube
app_launch_3: rumble
app_launch_4: prime-video
name_position: top
visible_name_text_color: '#ff0000'
scale: '120'

I also just tried this config and still no volume controls:
type: custom:firemote-card
entity: media_player.android_tv_192_168_3_46
device_family: nvidia-shield
device_type: shield-tv-2019
compatibility_mode: default
android_tv_remote_entity: remote.master_tv
visible_name_text: Master TV
defaultRemoteStyle_override: ''
app_launch_1: tivimate
app_launch_2: youtube
app_launch_3: rumble
app_launch_4: prime-video
name_position: top
visible_name_text_color: '#ff0000'
scale: '120'

For reference, CEC is enabled.

**Comment by unc2k3** [comment]
I do see that the volume buttons do work when using the Mushroom Media Player Card, so perhaps there is a way to override the volume buttons on the firemote with the controls from the Mushroom card?

Scratch that, I was on the wrong  one.  The volume buttons DO NOT work with the Mushroom card.

**Reply by PRProd** [reply]
How do you have your mushroom card attached to your shield?  ADB?  Android TV Remote?

**Reply by PRProd** [reply]
ha!  no worries... moving on

**Reply by unc2k3** [reply]
Not sure exactly.  I've got multiple entities for it.  It is setup with ADB and Android TV Remote
entity: media_player.android_tv_192_168_3_46
entity: media_player.master_tv
entity: media_player.master_shield

**Comment by PRProd** [comment]
@unc2k3 ~

I've seen some strange CEC behavior happen.  It usually comes down to one of three things:
 1. CEC isn't properly enabled EVERYWHERE on your media device (SHIELD in your case)
 2. CEC isn't properly enabled EVERYWHERE on your TV
 3. The HDMI cable is supposed to work properly to carry CEC, and it seems to have no issues with video, yet replacing the cable with a super high grade (4K or 8K) HDMI cable fixes the issue.

About 1 and 2 - Some devices and TVs have a main "turn CEC on/off" function, and then a separate "allow volume control" or "allow x y z etc. control" features in another place.  That's bitten me a few times.

**Reply by PRProd** [reply]
I also forgot to mention - If you're not running your HDMI cable directly between your media player device and your TV (for example, you use a switch, home theater, splitter or something else in between) that can cause CEC issues as well.

**Reply by unc2k3** [reply]
I'll have to do some digging.  It's a Samsung TV.  I noticed that if I enable Samsung IR in the Nvidia settings, it disables the Shield's Volume Control CEC toggle, and the volume on the Shield remote works fine.  If I toggle the Samsung IR switch to off (thus enabling the Volume Control CEC toggle), I then lose volume controls on my Shield Remote all together.  I also checked the actual TV settings and it appears HDMI-CEC is enabled. Idk.  Not the end of the world if I can't get volume controls, but it would be nice to figure out.  

To note:  I swapped the hdmi cable for a high speed one, and the cable is also directly plugged in from the shield to the tv.  Starting to think this has something to do with the TV and Shield not playing nicely more so than the firemote app.

**Reply by PRProd** [reply]
The added complication with IR is that your physical shield remote has an IR emitter on the front of it, so depending on your settings, it's probably using IR to send an IR signal to your TV.  Firemote cannot directly do IR, so it relies on CEC.

There ARE ways to make Home Assistant do IR, and if you get that working then using a Firemote button override will work around this issue for you... but that can get expensive if you don't already have a broadcom device or some other IR blaster.

>Starting to think this has something to do with the TV and Shield not playing nicely more so than the firemote app.

I tend to agree

**Reply by unc2k3** [reply]
Thanks for all your help today.  I'll keep messing with the tv and shield and see if I can sort it out, but feel free to close this thread as I don't think the issue is related to the firemote card at this point.

**Reply by PRProd** [reply]
No problem @unc2k3!  I'm happy to help!

Don't forget, I still have an open question for you over on https://github.com/PRProd/HA-Firemote/issues/489
